hadoop-plugin/src/main/groovy/com/linkedin/gradle/dependency/CheckDependencyTask.groovy

```diff
@@ -0,0 +1,202 @@
+   * @param dependencyPattern The dependencyPattern against which dependency should be checked
+   */
+  void handleInfo(Project project, Dependency dependency, DependencyPattern dependencyPattern) {
+    project.logger.info("INFO: ($dependency.group:$dependency.name:$dependency.version), ${dependencyPattern.getMessage()}");
+  }
+
+  /**
+   * Method to handle warning. Subclasses can override this method to handle warning in their own way.
+   * @param project The Gradle project
+   * @param dependency The dependency to check
+   * @param dependencyPattern The dependencyPattern against which dependency should be checked
+   */
+  void handleWarn(Project project, Dependency dependency, DependencyPattern dependencyPattern) {
+    project.logger.warn("WARN: ($dependency.group:$dependency.name:$dependency.version), ${dependencyPattern.getMessage()}");
+  }
+
+/**
+ * Returns true if the dependency matches with dependencyPattern
+ * @param dependency The dependency to match
+ * @param dependencyPattern The dependencyPattern against which dependency should be matched
+ * @return true or false based on whether match is found or not
+ */
+  Boolean dependencyMatchesPattern(Project project, Dependency dependency, DependencyPattern dependencyPattern) {
+
+    // If all of the group, name and version are not specified then return false
+    if (dependency.group == null && dependency.name == null && dependency.version == null) {
+      return false;
+    }
+
+    Boolean matchFound = true;
+
+    if (dependencyPattern.getGroupPattern() != null) {
+      matchFound = matchFound && dependency.getGroup().matches(dependencyPattern.getGroupPattern());
+    }
+    if (dependencyPattern.getNamePattern() != null) {
+      matchFound = matchFound && dependency.getName().matches(dependencyPattern.getNamePattern());
+    }
+    if (dependencyPattern.getVersionPattern() != null) {
+      matchFound = matchFound && dependency.getVersion().matches(dependencyPattern.getVersionPattern());
+    }
+    return matchFound;
+  }
+
+  /**
+   * Returns the set of all the dependencies which should be checked.
+   * Subclasses can override this method to return their own set of dependencies. By default we
+   * return all the dependencies.
+   * @param project The gradle project
+   * @return all dependencies of the project
+   */
+  Set<Dependency> getDependencies(Project project) {
+    Set<Dependency> dependencies = new HashSet<Dependency>();
+    project.getConfigurations().each {
+      configuration -> dependencies.addAll(configuration.getAllDependencies().asList())
+    }
+    return dependencies;
+  }
+
+  /**
+   * Creates and returns instance of DependencyPatterns class from dependencyXmlFile
+   * @param project The gradle project
+   * @return an instance of DependencyPatterns
+   */
+  DependencyPatterns getDependencyPatterns(Project project) {
+    JAXBContext jaxbContext = JAXBContext.newInstance(DependencyPatterns.class);
```

Comment:
With gradle XML seems to verbose.
Some ideas. 
Why not specify a map literal?  Something like [version: *, group: com.linkedin.*]  . You won't need to parse XML, it will read by Groovy as a Map. You'll not need to define marshaller unmarshaller classes. etc.